src/dotnet-aspnet-codegenerator/CodeGenCommandExecutor.cs

```diff
@@ -0,0 +1,103 @@
+using NuGet.Frameworks;
+
+namespace Microsoft.Extensions.CodeGeneration
+{
+    public class CodeGenCommandExecutor 
+    {
+        private ProjectContext _projectContext;
+        private string[] _codeGenArguments;
+        private string _configuration;
+        private string _nugetPackageDir;
+        private ILogger _logger;
+        
+        public CodeGenCommandExecutor(ProjectContext projectContext, string[] codeGenArguments, string configuration, string nugetPackageDir, ILogger logger)
+        {
+            if(projectContext == null) 
+            {
+                throw new ArgumentNullException(nameof(projectContext));
+            }
+            if(codeGenArguments == null)
+            {
+                throw new ArgumentNullException(nameof(codeGenArguments));
+            }
+            if(logger == null)
+            {
+                throw new ArgumentNullException(nameof(logger));
+            }
+            _projectContext = projectContext;
+            _codeGenArguments = codeGenArguments;
+            _configuration = configuration;
+            _logger = logger;
+            _nugetPackageDir = nugetPackageDir;
+        }
+        
+        public int Execute()
+        {
+            var serviceProvider = new ServiceProvider();
+            Directory.SetCurrentDirectory(_projectContext.ProjectDirectory);
+            AddFrameworkServices(serviceProvider, _projectContext, _nugetPackageDir);
+            AddCodeGenerationServices(serviceProvider);
+            var codeGenCommand = serviceProvider.GetService<CodeGenCommand>();
+            codeGenCommand.Execute(_codeGenArguments);
+            return 0;
+        }
+        
+        private void AddFrameworkServices(ServiceProvider serviceProvider, ProjectContext context, string nugetPackageDir)
+        {
+            var applicationEnvironment = new ApplicationEnvironment(context.RootProject.Identity.Name, context.ProjectDirectory);
+            serviceProvider.Add(typeof(IServiceProvider), serviceProvider);
+            serviceProvider.Add(typeof(ProjectContext), context);
+            serviceProvider.Add(typeof(Workspace), context.CreateWorkspace());
+            serviceProvider.Add(typeof(IApplicationEnvironment), applicationEnvironment);
+            serviceProvider.Add(typeof(ICodeGenAssemblyLoadContext), DefaultAssemblyLoadContext.CreateAssemblyLoadContext(nugetPackageDir));
+            serviceProvider.Add(typeof(ILibraryManager), new LibraryManager(context));
+            serviceProvider.Add(typeof(ILibraryExporter), new LibraryExporter(context, applicationEnvironment));
+        }
+
+        private void AddCodeGenerationServices(ServiceProvider serviceProvider)
+        {
+            if (serviceProvider == null)
+            {
+                throw new ArgumentNullException(nameof(serviceProvider));
+            }
+
+            //Ordering of services is important here
+            serviceProvider.Add(typeof(ILogger), _logger);
```

Comment:
I have been meaning to change to use Reporter. Will change the logger to use Reporter for now. 
Will remove the custom logger altogether post RC2.